handler/legacy_test.go

```diff
@@ -155,3 +156,84 @@ func TestRetrieveCorrectURI(t *testing.T) {
 		})
 	})
 }
+
+func TestTagSearchDataWithURITopics(t *testing.T) {
+	Convey("Given tagSearchDataWithURITopics", t, func() {
+		// Set up mock cache list
+		cacheList, err := cache.GetMockCacheList(ctx)
+		if err != nil {
+			t.Fatalf("Failed to get mock cache list: %v", err)
+		}
+
+		searchData := models.SearchDataImport{
+			UID:             "12345",
+			URI:             "/economy/environmentalaccounts",
+			DataType:        "dataset",
+			Edition:         "2021",
+			JobID:           "job-123",
+			SearchIndex:     "search-index",
+			CanonicalTopic:  "canonical-topic",
+			CDID:            "CDID-123",
+			DatasetID:       "dataset-123",
+			Keywords:        []string{"keyword1", "keyword2"},
+			MetaDescription: "meta description",
+			ReleaseDate:     "2021-01-01",
+			Summary:         "summary",
+			Title:           "title",
+			Topics:          []string{"1234"},
+			TraceID:         "trace-123",
+			Cancelled:       false,
+			Finalised:       false,
+			ProvisionalDate: "2021-01-02",
+			Published:       true,
+			Survey:          "survey",
+			Language:        "en",
+		}
+
+		Convey("When topics are initially empty", func() {
+			searchData.Topics = []string{}
+			updatedImporterEventData := tagSearchDataWithURITopics(ctx, searchData, cacheList.Topic)
+
+			Convey("Then the resulting topics should be as expected", func() {
+				expectedTopics := []string{"6734", "1834"}
+
+				So(updatedImporterEventData.Topics, ShouldHaveLength, len(expectedTopics))
+				So(updatedImporterEventData.Topics, ShouldContain, "1834")
+				So(updatedImporterEventData.Topics, ShouldContain, "6734")
+			})
+		})
+
+		Convey("When topics are initially not empty", func() {
+			updatedImporterEventData := tagSearchDataWithURITopics(ctx, searchData, cacheList.Topic)
+
+			Convey("Then the resulting topics should be as expected", func() {
```

Comment:
```suggestion
			Convey("Then the resulting topics should contain only the defaults", func() {
```

having non-descript (and duplicated) phrases doesn't help future-you understand what is being tested or which test has failed